### ISSUE TITLE
Playback error

### ISSUE THREAD

**Issue by MMMATRIX** [vfsfitvnm/ViMusic#1114]
### Steps to reproduce the bug

After restored backup file, songs does not play offline

### Expected behavior

It should to let me listen offline

### Actual behavior

It does not let me listen offline

### Screenshots/Screen recordings

Screenshot
![Screenshot_2023-07-09-21-35-31-324_it vfsfitvnm vimusic](https://github.com/vfsfitvnm/ViMusic/assets/112827868/79ff0589-e961-49cb-bea2-10153ca9723c)

### ViMusic version

0.5.4

### Android version

Android 12

**Comment by vfsfitvnm** [comment]
Yes, it looks like youtube changed something on their end. I still didn't have the chance to investigate

**Comment by MMMATRIX** [comment]
> Yes, it looks like youtube changed something on their end. I still didn't have the chance to investigate

I need to say that. I need to relisten song to play in offline...I forgot to mention that

**Comment by siggi1984** [comment]
> ### Steps to reproduce the bug
> After restored backup file, songs does not play offline
> 
> ### Expected behavior
> It should to let me listen offline
> 
> ### Actual behavior
> It does not let me listen offline
> 
> ### Screenshots/Screen recordings
> Screenshot ![Screenshot_2023-07-09-21-35-31-324_it vfsfitvnm vimusic](https://user-images.githubusercontent.com/112827868/252165455-79ff0589-e961-49cb-bea2-10153ca9723c.jpg)
> 
> ### Logs
> _No response_
> 
> ### ViMusic version
> 0.5.4
> 
> ### Android version
> Android 12
> 
> ### Additional information
> _No response_

As far as I know, the songs in the cache are not saved. That means after a restore of the database you can't listen to anything offline. You would have to listen to the songs again to be able to play them offline afterwards.

Therefore, I do not see any error in the reproduction here.

**Comment by Officialbabs** [comment]
Hope you didn't clear your cache or something?
Because it depends cache to play songs offline..

**Comment by WBRK-dev** [comment]
The music does not get backed up. You need to listen the music again after restore to cache it again.

**Comment by ghost** [comment]
Please add support to **download songs** in the app!

**Comment by siggi1984** [comment]
https://github.com/25huizengek1/ViMusic

Have a look at this fork. It doesn't have a download function, but it may still come.

**Comment by owenconnorz** [comment]
Or use this for downloads has more features then you can imagine also has spotify canvas and yt sync.

https://github.com/maxrave-dev/SimpMusic


![Screenshot_20240220_212431_SimpMusic.jpg](https://github.com/vfsfitvnm/ViMusic/assets/96526481/1c0caf7a-28d8-4acf-b5fc-c48a54aa63e6)
